packages/link-modules/src/loaders/container.ts

```diff
@@ -0,0 +1,40 @@
+import { BaseRepository, getPivotRepository } from "@repositories"
+import { PivotService, getModuleService } from "@services"
+
+import { LoaderOptions } from "@medusajs/modules-sdk"
+import {
+  InternalModuleDeclaration,
+  ModuleJoinerConfig,
+  ModulesSdkTypes,
+} from "@medusajs/types"
+import { asClass, asValue } from "awilix"
+
+export function containerLoader(entity, joinerConfig: ModuleJoinerConfig) {
+  return async (
+    {
+      options,
+      container,
+    }: LoaderOptions<
+      | ModulesSdkTypes.ModuleServiceInitializeOptions
+      | ModulesSdkTypes.ModuleServiceInitializeCustomDataLayerOptions
+    >,
+    moduleDeclaration?: InternalModuleDeclaration
+  ): Promise<void> => {
+    const [primary, foreign] = joinerConfig.relationships!
+
+    container.register({
+      joinerConfig: asValue(joinerConfig),
+      primaryKey: asValue(primary.foreignKey.split(",")),
+      foreignKey: asValue(foreign.foreignKey),
+      extraFields: asValue(
+        Object.keys(joinerConfig.databaseConfig?.extraFields || {})
+      ),
+
+      linkModuleService: asClass(getModuleService(joinerConfig)).singleton(),
+      pivotService: asClass(PivotService).singleton(),
```

Comment:
thought: I wonder if we should just stick to link* instead of pivot*, to not introduce another synonym in the link module world.

Reply:
I can rename that in this other [PR](https://github.com/medusajs/medusa/pull/4870) that will be merged after this one.